File: python/video_veo.py
"""
x402 API Example: Video Generation with Google Veo 3 Fast

This example shows premium video generation with Veo 3.
Higher quality than VideoGenAPI, but costs more.

Prerequisites:
1. pip install -r requirements.txt
2. python generate-wallet.js (get wallet address)
3. Fund wallet with USDC on Base
4. Create .env file with PRIVATE_KEY

Run: python python/video_veo.py
"""

from x402.clients.requests import x402_requests
from eth_account import Account
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Configuration
API_BASE_URL = os.getenv('API_BASE_URL', 'https://beatsx402.ai')
PRIVATE_KEY = os.getenv('PRIVATE_KEY')

if not PRIVATE_KEY:
    print('❌ Error: PRIVATE_KEY not found in .env file')
    print('\n📝 Setup:')
    print('1. Run: node generate-wallet.js')
    print('2. Copy .env.example to .env')
    print('3. Add your PRIVATE_KEY to .env\n')
    exit(1)


def main():
    try:
        print('🎬 x402 Veo 3 Fast Example\n')
        print('=' * 60)
        
        # 1. Set up wallet with x402
        account = Account.from_key(PRIVATE_KEY)
        session = x402_requests(account)
        
        print(f'\n💰 Wallet: {account.address}')
        print('💸 Cost: $2.00 USDC')
        print('🔗 Network: Base')
        print('🤖 Model: Google Veo 3 Fast')
        print('⭐ Quality: Premium')
        
        # 2. Generate video - PAID REQUEST ($2.00)
        print('\n📤 Generating video...\n')
        
        response = session.post(
            f'{API_BASE_URL}/v1/veo3/fast/generate',
            json={
                'prompt': 'Epic ocean waves crashing with dramatic lightning storm approaching',
                'aspect_ratio': '16:9',
                'duration': '8s',
                'resolution': '1080p',
                'generate_audio': True
            }
        )
        
        data = response.json()
        
        # 3. Display response
        print('✅ Video generated!\n')
        print('=' * 60)
        
        if 'video_url' in data:
            print('🎬 Video URL:')
            print(data['video_url'])
            
            if 'duration' in data:
                print(f"\n📋 Duration: {data['duration']}")
            if 'resolution' in data:
                print(f"📺 Resolution: {data['resolution']}")
        else:
            print(str(data))
        
        print('=' * 60)
        
        print('\n💡 How x402 worked:')
        print('1. ✅ Initial request sent → Server: 402 Payment Required')
        print('2. ✅ x402 client signed $2.00 USDC payment on Base')
        print('3. ✅ Retried with X-PAYMENT header')
        print('4. ✅ Server verified payment & processed request')
        print('5. ✅ Payment settled in ~2 seconds\n')
        
        print('💡 Note: Veo 3 Fast is premium quality')
        print('   - Better motion and coherence than VideoGenAPI')
        print('   - Supports audio generation')
        print('   - 4-8 second videos at 720p or 1080p\n')
        
    except Exception as error:
        print(f'❌ Error: {str(error)}')
        
        if '402' in str(error):
            print('\n💡 Payment issue - check wallet balance:')
            print('   - Ensure wallet has USDC on Base network')
            print('   - Minimum: $2.00 USDC for this request\n')


if __name__ == '__main__':
    main()

File: javascript/video-veo.js
/**
 * x402 API Example: Video Generation with Google Veo 3 Fast
 * 
 * This example shows premium video generation with Veo 3.
 * Higher quality than VideoGenAPI, but costs more.
 * 
 * Prerequisites:
 * 1. npm install
 * 2. node generate-wallet.js (get wallet address)
 * 3. Fund wallet with USDC on Base
 * 4. Create .env file with PRIVATE_KEY
 * 
 * Run: node javascript/video-veo.js
 */

import { wrapFetchWithPayment } from 'x402-fetch';
import { createWalletClient, http } from 'viem';
import { privateKeyToAccount } from 'viem/accounts';
import { base } from 'viem/chains';
import 'dotenv/config';

// Configuration
const API_BASE_URL = process.env.API_BASE_URL || 'https://beatsx402.ai';
const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error('❌ Error: PRIVATE_KEY not found in .env file');
  console.log('\n📝 Setup:');
  console.log('1. Run: node generate-wallet.js');
  console.log('2. Copy .env.example to .env');
  console.log('3. Add your PRIVATE_KEY to .env\n');
  process.exit(1);
}

async function main() {
  try {
    console.log('🎬 x402 Veo 3 Fast Example\n');
    console.log('='.repeat(60));
    
    // 1. Set up wallet with x402
    const account = privateKeyToAccount(PRIVATE_KEY);
    const walletClient = createWalletClient({
      account,
      chain: base,
      transport: http()
    });
    const fetchWithPayment = wrapFetchWithPayment(fetch, walletClient);
    
    console.log(`\n💰 Wallet: ${account.address}`);
    console.log('💸 Cost: $2.00 USDC');
    console.log('🔗 Network: Base');
    console.log('🤖 Model: Google Veo 3 Fast');
    console.log('⭐ Quality: Premium');
    
    // 2. Generate video - PAID REQUEST ($2.00)
    console.log('\n📤 Generating video...\n');
    
    const response = await fetchWithPayment(`${API_BASE_URL}/v1/veo3/fast/generate`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        prompt: 'Epic ocean waves crashing with dramatic lightning storm approaching',
        aspect_ratio: '16:9',
        duration: '8s',
        resolution: '1080p',
        generate_audio: true
      })
    });
    
    const data = await response.json();
    
    // 3. Display response
    console.log('✅ Video generated!\n');
    console.log('='.repeat(60));
    
    if (data.video_url) {
      console.log('🎬 Video URL:');
      console.log(data.video_url);
      
      if (data.duration) {
        console.log(`\n📋 Duration: ${data.duration}`);
      }
      if (data.resolution) {
        console.log(`📺 Resolution: ${data.resolution}`);
      }
    } else {
      console.log(JSON.stringify(data, null, 2));
    }
    
    console.log('='.repeat(60));
    
    console.log('\n💡 How x402 worked:');
    console.log('1. ✅ Initial request sent → Server: 402 Payment Required');
    console.log('2. ✅ x402-fetch signed $2.00 USDC payment on Base');
    console.log('3. ✅ Retried with X-PAYMENT header');
    console.log('4. ✅ Server verified payment & processed request');
    console.log('5. ✅ Payment settled in ~2 seconds\n');
    
    console.log('💡 Note: Veo 3 Fast is premium quality');
    console.log('   - Better motion and coherence than VideoGenAPI');
    console.log('   - Supports audio generation');
    console.log('   - 4-8 second videos at 720p or 1080p\n');
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    
    if (error.message.includes('402')) {
      console.log('\n💡 Payment issue - check wallet balance:');
      console.log('   - Ensure wallet has USDC on Base network');
      console.log('   - Minimum: $2.00 USDC for this request\n');
    }
  }
}

main();

